src/services/AbortRequestProcessor.ts

```diff
@@ -0,0 +1,79 @@
+import { Metrics, MetricUnits } from "@aws-lambda-powertools/metrics";
+import { Logger } from "@aws-lambda-powertools/logger";
+import { BavService } from "./BavService";
+import { AuthSessionState } from "../models/enums/AuthSessionState";
+import { HttpCodesEnum } from "../models/enums/HttpCodesEnum";
+import { MessageCodes } from "../models/enums/MessageCodes";
+import { TxmaEventNames } from "../models/enums/TxmaEvents";
+import { ISessionItem } from "../models/ISessionItem";
+import { EnvironmentVariables } from "../utils/Constants";
+import { absoluteTimeNow } from "../utils/DateTimeUtils";
+import { createDynamoDbClient } from "../utils/DynamoDBFactory";
+import { checkEnvironmentVariable } from "../utils/EnvironmentVariables";
+import { Response } from "../utils/Response";
+import { buildCoreEventFields } from "../utils/TxmaEvent";
+
+export class AbortRequestProcessor {
+  private static instance: AbortRequestProcessor;
+
+  private readonly logger: Logger;
+
+	private readonly issuer: string;
+
+	private readonly txmaQueueUrl: string;
+
+  private readonly metrics: Metrics;
+
+  private readonly BavService: BavService;
+
+  constructor(logger: Logger, metrics: Metrics) {
+  	this.issuer = checkEnvironmentVariable(EnvironmentVariables.ISSUER, logger);
+  	this.txmaQueueUrl = checkEnvironmentVariable(EnvironmentVariables.TXMA_QUEUE_URL, logger);
+  	const sessionTableName: string = checkEnvironmentVariable(EnvironmentVariables.SESSION_TABLE, logger);
+
+  	this.logger = logger;
+  	this.metrics = metrics;
+  	this.BavService = BavService.getInstance(sessionTableName, this.logger, createDynamoDbClient());
+  }
+
+  static getInstance(logger: Logger, metrics: Metrics): AbortRequestProcessor {
+  	if (!AbortRequestProcessor.instance) {
+  		AbortRequestProcessor.instance = new AbortRequestProcessor(logger, metrics);
+  	}
+  	return AbortRequestProcessor.instance;
+  }
+
+  async processRequest(sessionId: string): Promise<Response> {
+  	const session = await this.BavService.getSessionById(sessionId);
+
+  	if (!session) {
+  		this.logger.error("No session found for session id", {
+  			messageCode: MessageCodes.SESSION_NOT_FOUND,
+  		});
+  		return new Response(HttpCodesEnum.UNAUTHORIZED, `No session found with the session id: ${sessionId}`);
+  	}
+
+  	this.logger.appendKeys({
+  		govuk_signin_journey_id: session?.clientSessionId,
```

Comment:
Do we also want the sessionId?

Reply:
Gonna leave it out only cause we shouldn't be accessing the DB anyways and IF we need to, we can search with the gov ID